sefaz/accesskey/validation.go

```diff
@@ -5,14 +5,54 @@ import (
 	"github.com/arquivei/foundationkit/sefaz/stakeholder"
 )
 
+// Check checks if an access key is valid. It returns an error with an specific code based on
+// the validation problem
+func Check(accessKey AccessKey) error {
+	const op errors.Op = "accesskey.Check"
+
+	if accessKey == "" {
+		return errors.E(op, ErrEmptyAccessKey, ErrCodeEmptyAccessKey)
+	}
+
+	if len(accessKey) != 44 {
+		return errors.E(op, ErrInvalidLength, ErrCodeInvalidLength)
+	}
+
+	if !isDigitOnly(accessKey) {
+		return errors.E(op, ErrInvalidCharacter, ErrCodeInvalidCharacter)
+	}
+
+	if !isValidUF(accessKey[0:2].String()) {
+		return errors.E(op, ErrInvalidUF, ErrCodeInvalidUF)
+	}
+
+	if !isValidMonth(accessKey[4:6].String()) {
+		return errors.E(op, ErrInvalidMonth, ErrCodeInvalidMonth)
+	}
+
+	if !isValidCPFCNPJ(accessKey[6:20].String()) {
+		return errors.E(op, ErrInvalidCPFCNPJ, ErrCodeInvalidCPFCNPJ)
+	}
+
+	if !isValidModel(accessKey[20:22].String()) {
+		return errors.E(op, ErrInvalidModel, ErrCodeInvalidModel)
+	}
+
+	if !isValidationDigitCorrect(accessKey.String()) {
+		return errors.E(op, ErrInvalidDigit, ErrCodeInvalidDigit)
+	}
+
+	return nil
+}
+
 // validate execute all sub-routines necessary to perform a full accesskey validation
 func validate(accessKey AccessKey) error {
 	const op errors.Op = "validate"
 	if accessKey == "" {
 		return errors.E(op, ErrEmptyAccessKey, ErrCodeEmptyAccessKey)
 	}
 	if len(accessKey) != 44 {
-		return errors.E(op, ErrInvalidLenght, ErrCodeInvalidAccessKey)
+		return errors.E(op, ErrInvalidLength, ErrCodeInvalidAccessKey)
 	}
 
 	if !isDigitOnly(accessKey) {
```

Comment:
Nit: the old code could use the new `check` function, and override the returned `error.Code` to keep the old behavior

Reply:
Good idea

Reply:
Just a point, I actually did it the other way around. Once the new function has a good test coverage, it makes more sense to make the changes there and keep the old code unchanged.